### ISSUE TITLE
Emits prioritizing the wrong type

### ISSUE THREAD

**Issue by sheremet-va** [johnsoncodehk/volar#188]
`@submit` and all 'standard' events are always `(payload: Event) => void | undefined` even if I overwrite them inside `emits` option.

Reproduction:
```vue
// child
<script>
import { defineComponent } from 'vue';

export default defineComponent({
   emits: {
       sumbit: (id: number) => true
   }
});
</script>
```

```vue
// parent
<script>
import { defineComponent } from 'vue';
import Child from './Child';

export default defineComponent({
   components: {
      Child
   },
  setup() {
     return {
        test: (id: number) => console.log(id)     
    }
  }
});
</script>

<template>
    <child @submit="test" />
</template>
```

This also broke `vue-tsc`!

Related issue: `vue-tsc` throws error if `emits` defined with array (`emits: ['remove']`):

![image](https://user-images.githubusercontent.com/16173870/118618994-dc2bc700-b7cc-11eb-945e-16840a11860e.png)

**Comment by sheremet-va** [comment]
Also I can't rollback to previous version (I try installing `0.1.0`, which I had before, but I still have errors)

**Comment by johnsoncodehk** [comment]
Sorry about that, I will fixed `vscode-vue-languageservice` version in `vue-tsc` to avoid `vue-tsc` broke with `vscode-vue-languageservice` update.

**Comment by sheremet-va** [comment]
Thank you. Before that I rolled back to previous package-lock version and it helped (so finally I know why it exists 😆 ).

**Comment by johnsoncodehk** [comment]
Fixed in 0.25.6 and update with vue-tsc 0.1.4.
